src/components/Table/types.ts

```diff
@@ -0,0 +1,13 @@
+import { ReactNode } from "react"
+
+export interface WithID {
+  id: string
+}
+
+export interface ColumnConfig<Datum extends WithID> {
+  label: string | null
+  getValue?: (datum: Datum) => ReactNode
+  sortable?: boolean
+}
+
+export type ColumnConfigs<Datum extends WithID> = ColumnConfig<Datum>[]
```

Comment:
Also, not sure how I feel about exporting collection types (`ColumnConfigs`, a collection of `ColumnConfig`).
It's definitely not wrong, but also not common practice, I don't know why. 🤔 

Is it assumed that making a collection is trivial? Not worth the extra characters in the bundle? Maybe the user wants to name the type something else? `ColumnConfigList`? None of these arguments seem strong enough to me.